tree_test.go

```diff
@@ -1322,13 +1322,20 @@ func TestManipulateChildren(t *testing.T) {
 }
 
 func TestLeafNodeInsert(t *testing.T) {
-	values := make([][]byte, NodeWidth)
 	valIdx := 42
-	values[valIdx] = testValue
-	ln := NewLeafNode(ffx32KeyTest[:StemSize], values)
```

Comment:
We can't create "direct leaf node" now since all the internal data is lazily computed. This caused a crash later when doing `ln.Hash()` called `(*LeafNode).Commitment()`, which panicked since there wasn't any calculated commitment.

The solution is always to use the tree from a properly created tree, since before inspecting internals of things you should always call `tree.Commitment()` to do all the pending calculations.

Another solution is making `(*LeafNode).Commitment()` not panicked if `commitment == nil`, so compute it if that's the case. Technically speaking, that would be my preferred solution since we shouldn't have panicked but considering that case should never happen...